src/cryptoadvance/specter/notifications/notifications.py

```diff
@@ -0,0 +1,101 @@
+import logging
+
+logger = logging.getLogger(__name__)
+import datetime
+import hashlib
+
+
+class NotificationTypes:
+    debug = "debug"
+    information = "information"
+    warning = "warning"
+    error = "error"
+    exception = "exception"
+
+
+class Notification(dict):
```

Comment:
I had it as attributes in [Notifications](https://github.com/relativisticelectron/specter-desktop/blob/f48642ac768603f32f11f6157e87262367149686/src/cryptoadvance/specter/notifications/notifications.py#L16) at first, but I changed it to a dict. I try to lay out the reasoning.

There are 3 notification data structures:
1. [Notifications](https://github.com/relativisticelectron/specter-desktop/blob/f48642ac768603f32f11f6157e87262367149686/src/cryptoadvance/specter/notifications/notifications.py#L16)  (currently derived from a dict)
2. [js_notification data structure](https://github.com/relativisticelectron/specter-desktop/blob/f48642ac768603f32f11f6157e87262367149686/src/cryptoadvance/specter/notifications/notifications.py#L84)  which is passed via  [get_new_notifications](https://github.com/relativisticelectron/specter-desktop/blob/f48642ac768603f32f11f6157e87262367149686/src/cryptoadvance/specter/server_endpoints/wallets_api.py#L103) to javascript. This datastructure is identical to the [Notifications_API datastructure](https://developer.mozilla.org/en-US/docs/Web/API/Notifications_API/Using_the_Notifications_API) but has additional fields, if needed.
3. [Notifications_API datastructure](https://developer.mozilla.org/en-US/docs/Web/API/Notifications_API/Using_the_Notifications_API) which is passed to [webapi_notification](https://developer.mozilla.org/en-US/docs/Web/API/Notifications_API/Using_the_Notifications_API)


I had that at first, but I changed it to a dict because:
- The conversion of [Notifications](https://github.com/relativisticelectron/specter-desktop/blob/f48642ac768603f32f11f6157e87262367149686/src/cryptoadvance/specter/notifications/notifications.py#L16) --> [js_notification data structure](https://github.com/relativisticelectron/specter-desktop/blob/f48642ac768603f32f11f6157e87262367149686/src/cryptoadvance/specter/notifications/notifications.py#L84)  in  [to_js_notification](https://github.com/relativisticelectron/specter-desktop/blob/f48642ac768603f32f11f6157e87262367149686/src/cryptoadvance/specter/notifications/notifications.py#L84)  is much simpler and doesn't need changes when additional attributes (which would be in `options`) are added 
- Attribute need a custom __str__ method, so I can hash it. Adding any attributes then also need a change of that one.
- All dicts made it more uniform for me to switch between javascript (dict format) and python

I would be OK with switching it back to attributes, though.    Should I?

Reply:
Oh, i'm not proposing attributes in the Notification class. I thought about a class with attributes and maybe properties. I understand the reasoning that it's easier to convert back and forth with javascript but that ussually also makes troubleshooting much more complicated. Javascript is especially untyped and because that brings its problems, people invented typescript and json-schema to have at least a bit more control on what's going on and raise issues where they occur and not create weird issue which cover the issue because they got passed multiple times through code undiscovered because all is "generic".

So would be cool to make it more explicit as this would, even if more code is needed, enable explicit error-handling and a better idea (spec) what messages are exactly and what they are suppose to look like and don't look like.

Reply:
> Oh, i'm not proposing attributes in the Notification class. I thought about a class with attributes and maybe properties.

I am confused.   In the [past](https://github.com/relativisticelectron/specter-desktop/blob/56446bd5108a8114e8652abc02a64744fe1ed180/src/cryptoadvance/specter/notifications/notifications.py#L19) I had attributes:
```
class Notification:
    "A Notification is a datastructure to store title, body, ..."

    def __init__(
        self, title, notification_type=None, body=None, target_uis="all", **kwargs
    ):
        self.title = title
        self.date = datetime.datetime.now()
        self.deleted = False
        self.first_read = None
        self.body = body
        self.icon = None
        self.timeout = None
```

Is this what you had in mind?

Reply:
Ok. Yes your are right. (Especially now that the code is changing slower)